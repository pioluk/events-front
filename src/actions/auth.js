/* @flow */

import {
  LOGIN_REQUEST,
  LOGIN_SUCCESS,
  LOGIN_FAILURE,
  LOGOUT
} from '../constants/auth'

export function loginRequest (username: string, password: string) {
  return {
    type: LOGIN_REQUEST,
    username,
    password
  }
}

export function loginSuccess (user: string, token: string) {
  return {
    type: LOGIN_SUCCESS,
    user,
    token
  }
}

export function loginFailure (error: Error) {
  return {
    type: LOGIN_FAILURE,
    error
  }
}

export function logout () {
  return {
    type: LOGOUT
  }
}
