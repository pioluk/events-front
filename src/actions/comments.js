// @flow

import {
  FETCH_COMMENTS,
  FETCH_COMMENTS_SUCCESS,
  FETCH_COMMENTS_FAILURE,
  ADD_COMMENT,
  ADD_COMMENT_SUCCESS,
  ADD_COMMENT_FAILURE,
  REMOVE_COMMENT,
  REMOVE_COMMENT_SUCCESS,
  REMOVE_COMMENT_FAILURE,
} from '../constants/comments'

export function fetchComments (eventId: number, limit: number, offset: number) {
  return {
    type: FETCH_COMMENTS,
    eventId,
    limit,
    offset
  }
}

export function fetchCommentsSuccess (count: number, comments: any[]) {
  return {
    type: FETCH_COMMENTS_SUCCESS,
    count,
    comments
  }
}

export function fetchCommentsFailure (error: Error) {
  return {
    type: FETCH_COMMENTS_FAILURE,
    error
  }
}

export function addComment (text: string) {
  return {
    type: ADD_COMMENT,
    text
  }
}

export function addCommentSuccess (comment: any) {
  return {
    type: ADD_COMMENT_SUCCESS,
    comment
  }
}

export function addCommentFailure (error: Error) {
  return {
    type: ADD_COMMENT_FAILURE,
    error
  }
}

export function removeComment (id: number) {
  return {
    type: REMOVE_COMMENT,
    id
  }
}

export function removeCommentSuccess (id: number) {
  return {
    type: REMOVE_COMMENT_SUCCESS,
    id
  }
}

export function removeCommentFailure (error: Error) {
  return {
    type: REMOVE_COMMENT_FAILURE,
    error
  }
}
