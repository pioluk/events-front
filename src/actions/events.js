import {
  FETCH_EVENTS_REQUEST,
  FETCH_EVENTS_SUCCESS,
  FETCH_EVENTS_FAILURE
} from '../constants/events'

export function fetchEventsRequest () {
  return {
    type: FETCH_EVENTS_REQUEST
  }
}

export function fetchEventsSuccess (events) {
  return {
    type: FETCH_EVENTS_SUCCESS,
    events
  }
}

export function fetchEventsFailure (error) {
  return {
    type: FETCH_EVENTS_FAILURE,
    error
  }
}
