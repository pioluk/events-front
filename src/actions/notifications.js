import { DISPLAY_NOTIFICATION } from '../constants/notifications'

export function displayNotification (message) {
  return {
    type: DISPLAY_NOTIFICATION,
    message
  }
}
