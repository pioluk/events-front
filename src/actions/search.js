// @flow

import {
  SEARCH_EVENTS,
  SEARCH_EVENTS_RESET,
  SEARCH_EVENTS_SUCCESS
} from '../constants/search'

export function searchEvents (page: number, query: string) {
  return {
    type: SEARCH_EVENTS,
    page,
    query
  }
}

export function searchEventsReset () {
  return {
    type: SEARCH_EVENTS_RESET
  }
}

export function searchEventsSuccess (count: number, events: Array<any>) {
  return {
    type: SEARCH_EVENTS_SUCCESS,
    count,
    events
  }
}
