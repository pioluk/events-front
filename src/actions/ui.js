export function selectEvent(color, image) {
  return {
    type: 'SELECT_EVENT',
    color,
    image
  }
}
