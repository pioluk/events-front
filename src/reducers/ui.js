/* @flow */

type UiState = {
  selectedEventColor: ?string,
  selectedEventImage: ?string
}

const initialState: UiState = {
  selectedEventColor: null,
  selectedEventImage: null
}

export default function uiReducer (state: UiState = initialState, action: any): UiState {
  switch (action.type) {
    case 'SELECT_EVENT':
      return {
        selectedEventColor: action.color,
        selectedEventImage: action.image
      }

    default:
      return state
  }
}
