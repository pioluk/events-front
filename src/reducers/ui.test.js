// global describe,test,expect

import reducer from './ui'
import * as actions from '../actions/ui'

describe('ui reducer', () => {
  test('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  test('should handle SELECT_EVENT', () => {
    const color = '#a5aaf2'
    const image = 'ec8672d6-3787-4ce6-a35b-715d012f23e4.jpg'
    const state = reducer(undefined, actions.selectEvent(color, image))
    expect(state).toMatchSnapshot()
  })
})
