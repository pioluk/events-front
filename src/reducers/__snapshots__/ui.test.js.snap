exports[`ui reducer should handle SELECT_EVENT 1`] = `
Object {
  "selectedEventColor": "#a5aaf2",
  "selectedEventImage": "ec8672d6-3787-4ce6-a35b-715d012f23e4.jpg",
}
`;

exports[`ui reducer should return the initial state 1`] = `
Object {
  "selectedEventColor": null,
  "selectedEventImage": null,
}
`;
